### PR TITLE
move Dispose calls to finally blocks

### PR DESCRIPTION
**Pull request by honfika**
Doneness:
- [X] Build is okay - I made sure that this change is building successfully.
- [X] No Bugs - I made sure that this change is working properly as expected. It doesn't have any bugs that you are aware of. 
- [X] Branching - If this is not a hotfix, I am making this request against develop branch

### CODE REVIEW

File: Titanium.Web.Proxy/RequestHandler.cs [hunk 1]
```diff
@@ -271,21 +259,17 @@ partial class ProxyServer
         }
 
 
-        private async Task<bool> HandleHttpSessionRequestInternal(TcpConnection connection,
-            SessionEventArgs args, bool closeConnection)
+        private async Task<bool> HandleHttpSessionRequestInternal(TcpConnection connection, SessionEventArgs args, bool closeConnection)
         {
+            bool dispose = true;
```
Comment (justcoding121) [inline Titanium.Web.Proxy/RequestHandler.cs]: should be disposed? I think this should be false by default and then if needs to be disposed we will set it to true. I see some inconsistency in this method.

Comment (honfika) [inline Titanium.Web.Proxy/RequestHandler.cs]: The other methods have "disposed" flag, this is "dispose" which means that Dispose should be called. So this is not the return value, which is true when the streams were disposed.
I'll try to make it similar to the other methods, please wait for the next commit.

**Comment by honfika** [comment]
Is this better now?

**Comment by justcoding121** [comment]
Thanks looks better now

**Pull request merged**